pvlib/iotools/tmy.py

```diff
@@ -198,9 +214,16 @@ def read_tmy3(filename, coerce_year=None, recolumn=True):
     # NOTE: as of pvlib-0.6.3, min req is pandas-0.18.1, so pd.to_timedelta
     # unit must be in (D,h,m,s,ms,us,ns), but pandas>=0.24 allows unit='hour'
     data.index = data_ymd + pd.to_timedelta(shifted_hour, unit='h')
-
-    if recolumn:
+    if map_variables is None:
         data = _recolumn(data)  # rename to standard column names
+        warnings.warn(
+            'TMY3 variable names will be renamed to pvlib conventions by '
+            'default starting in pvlib 0.9.5. Specify map_variables=True '
```

Comment:
Version 0.9.5 will be the next release and there we will not rename by default but rather raise a warning as you are implementing there.

The release where we will change the default of ``map_variables`` from None to True will have to be further down the road, e.g., 0.10.0. Perhaps @kanderso-nrel can provide a suggestion for what release that might be?

Reply:
Pardon for late response, for the test case that checks if the warning is raised properly should I modify the already present test_read_tmy3_recolumn function.

Reply:
I'm not completely sure what you're asking, but the test that you have written looks good to me :)